tests/test_streaming.py

```diff
@@ -0,0 +1,16 @@
+import unittest
```

Comment:
Have you thought about [pytest](https://docs.pytest.org/en/latest/)? It's backwards-compatible with unittest, but the style is a bit different (e.g., no special asserts), and has things like fixtures, plugins, etc. While `unittest` is in the standard library (and can't be removed because of backwards compatibility) the community direction seems to be towards using pytest for new code. 

That said, the LSST style [requires unittest](https://developer.lsst.io/python/testing.html) for historical reasons, so if you don't want to context-switch between two styles depending on which repo you're using unittest would make sense.

I'm fine with it either way -- just bringing it up in case you didn't know about pytest.

Reply:
I decided to go with unittest because I found it easier to navigate its API. https://docs.pytest.org/en/latest/reference.html is _56 pages long_, and the package just does so much. It has its own plugin architecture and a system of hooks, it has its own facilities for tempfiles, it has ways to capture file descriptors, and it rewrites compiled code - it's just an awful lot. I wanted something straightforward and simple, something that would be obvious in its behavior, and unittest seemed like a better match (though it's still more than I want! sometimes, I think folks forget that tests are just code like any other...).